client/state/atomic-transfer/hooks/use-atomic-transfer-query.ts

```diff
@@ -0,0 +1,61 @@
+import { useQuery } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+import { transferStates, TransferStates } from 'calypso/state/automated-transfer/constants';
+import { SiteSlug } from 'calypso/types';
+
+interface ErrorResponse {
+	code: 'no_transfer_record';
+}
+
+interface SuccessResponse {
+	status: TransferStates;
+}
+
+const fetchLatestAtomicTransfer = (
+	siteSlug: SiteSlug
+): Promise< ErrorResponse | SuccessResponse > =>
+	wpcom.req.get( {
+		path: `/sites/${ siteSlug }/atomic/transfers/latest`,
+		apiNamespace: 'wpcom/v2',
+	} );
+
+interface UseAtomicTransferQueryOptions {
+	refetchInterval?: number;
+}
+
+const endStates = [
+	transferStates.NONE,
+	transferStates.COMPLETE,
+	transferStates.FAILURE,
+	transferStates.ERROR,
+	transferStates.REVERTED,
+] as const;
+
+export const useAtomicTransferQuery = (
+	siteSlug: SiteSlug,
+	{ refetchInterval }: UseAtomicTransferQueryOptions
+) => {
+	const { data } = useQuery( [ 'sites', siteSlug, 'atomic', 'transfers', 'latest' ], {
```

Comment:
Maybe a nit pick, but what about a cache key like `['sites', siteSlug, 'atomic-transfers', 'latest']`

I think it makes more sense when deciding how the cache should be invalidated.
Invalidating `['sites',siteSlug,'atomic-transfers']` would invalidate all transfers, whether it's the latest or not
Using the key as written, what would invalidating `['sites', siteSlug, 'atomic']` mean? Invalidate transfer state, GitHub state, SSH keys, all at the same time?